### PR TITLE
Public api

### PR DESCRIPTION
**Pull request by ghislainfourny**
This is the first, basic version of the Public API for those who want to use it via maven.

You can see the javadoc here: http://rumbledb.org/site/apidocs/org/rumbledb/api/package-summary.html

I moved over the Item class to this API and documented it with Javadoc. Most files modified are only a change of the imported package for the Item class.
 
In the future, we'll allow passing input to the dynamic context, and in the more remote future, binding an input variable to an RDD provided by the user.

### CODE REVIEW

**Comment by ghislainfourny** [comment]
If you try to generate the javadocs with mvn javadoc:javadoc, there will be an error about the builtin Java modules, which is a bug of the maven javadoc plugin. But the files still get generated properly.

File: src/main/java/org/rumbledb/api/Item.java [hunk 1]
```diff
@@ -109,96 +119,210 @@ public int compareTo(Item other) {
...
+	 */
     public boolean isAtomic() {
         return false;
     }
 
-    public boolean isNumber() {
-        return false;
-    }
-
+    /**
```
Comment (pjfanning) [inline src/main/java/org/rumbledb/api/Item.java]: minor issue with tabs appearing the javadoc

Comment (ghislainfourny) [reply src/main/java/org/rumbledb/api/Item.java]: Good catch, this is fixed.

**Comment by ghislainfourny** [comment]
Thanks!

I think we can release the public API publicly (as a minor update Rumble 1.1.1) together with the maven central entry once you are ready.

**Pull request merged**